app/src/androidTest/java/com/github/se/icebreakrr/ui/sections/FilterScreenTest.kt

```diff
@@ -404,6 +407,112 @@ class FilterScreenTest {
+  fun testRadiusSliderInitialValue() {
+    val initialRadius = 100
+    var currentRadius = initialRadius
+
+    composeTestRule.setContent {
+      RadiusSlider(
+          radius = currentRadius, onDistanceChange = { newRadius -> currentRadius = newRadius })
+    }
+
+    // Verify the initial value displayed
+    composeTestRule
+        .onNodeWithTag("RadiusSliderTitle")
+        .assertExists()
+        .assertIsDisplayed()
+        .assertTextEquals("Search Radius: $initialRadius m")
+  }
+
+  @Test
+  fun testRadiusSliderChanges() {
+    val allowedValues = listOf(50, 100, 200, 300, 400, 500)
+    val currentRadius = mutableStateOf(100) // Use mutableStateOf here
+
+    composeTestRule.setContent {
+      RadiusSlider(
+          radius = currentRadius.value,
+          onDistanceChange = { newRadius -> currentRadius.value = newRadius })
+    }
+
+    allowedValues.forEachIndexed { index, expectedValue ->
+      // Perform slider action
+      composeTestRule.onNodeWithTag("RadiusSlider").performSemanticsAction(
+          SemanticsActions.SetProgress) {
+            it(index.toFloat()) // Move to specific index
+      }
+
+      // Verify the state updates
+      assertEquals(expectedValue, currentRadius.value)
+
+      // Verify the displayed text matches the value
+      composeTestRule
+          .onNodeWithTag("RadiusSliderTitle")
+          .assertExists()
+          .assertTextEquals("Search Radius: $expectedValue m")
+    }
+  }
+
+  @Test
+  fun testRadiusSliderBounds() {
+    val initialRadius = 50
+    val allowedValues = listOf(50, 100, 200, 300, 400, 500)
+    var currentRadius = initialRadius
+
+    composeTestRule.setContent {
+      RadiusSlider(
+          radius = currentRadius, onDistanceChange = { newRadius -> currentRadius = newRadius })
+    }
+
+    // Try setting progress outside the allowed range and verify it stays within bounds
+    composeTestRule.onNodeWithTag("RadiusSlider").performSemanticsAction(
+        SemanticsActions.SetProgress) {
+          it(-1f)
+        }
+    assertEquals(allowedValues.first(), currentRadius)
+
+    composeTestRule.onNodeWithTag("RadiusSlider").performSemanticsAction(
+        SemanticsActions.SetProgress) {
+          it(allowedValues.size.toFloat())
+        }
+    assertEquals(allowedValues.last(), currentRadius)
+  }
+
+  @Test
+  fun testRadiusSliderIntegration() {
+    val initialRadius = 100
+    filterViewModel.setSelectedRadius(initialRadius)
+
+    composeTestRule.setContent {
+      FilterScreen(navigationActions = navigationActionsMock, filterViewModel = filterViewModel)
+    }
+
+    // Verify the initial radius value is displayed
+    composeTestRule
+        .onNodeWithTag("RadiusSliderTitle", useUnmergedTree = true) // Updated
+        .assertExists()
+        .assertTextEquals("Search Radius: $initialRadius m")
+
+    // Change the slider value and verify
+    val newRadius = 300
+    val newIndex = listOf(50, 100, 200, 300, 400, 500).indexOf(newRadius)
```

Comment:
To make testing more efficient you can put this list at the top as a const or share it with the FilterScreen to allow compatibility

Reply:
I agree !